lsp-go.el

```diff
@@ -50,17 +50,17 @@ completing function calls."
 (defcustom lsp-gopls-build-flags ["-tags"]
   "A vector of flags passed on to the build system when invoked,
   applied to queries like `go list'."
-  :type '(repeat string)
+  :type '(vector string)
```

Comment:
there is no vector type, can you change this to restricted expression?

Reply:
Agree.

Reply:
Hi! I'm happy to change it. That said, there _is_ a vector type (at least, the docs list one): https://www.gnu.org/software/emacs/manual/html_node/elisp/Composite-Types.html#Composite-Types

```
(vector element-types...)
    Like list except that the value must be a vector instead of a list. The elements work the same as in list. 
```

I will be guided by y'all. Do you prefer `vector`, or restricted expression?

Reply:
You are right. I guess in the past we have used vector without a type parameter. Merging as it is.

Reply:
IIUC, `vector` is fixed length type. i.e. `(vector string)` is a vector of one string element.

> (list element-types...)
>    The value must be a list with exactly as many elements as the element-types given; and each element must fit the corresponding element-type.
>
>    For example, (list integer string function) describes a list of three elements; the first element must be an integer, the second a string, and the third a function. 

> (vector element-types...)
>    Like list except that the value must be a vector instead of a list. The elements work the same as in list.

Reply:
@iquiw thank you. I remember that there was a reason not to use the vector type but I didn't remember what was the case.

Reply:
Oh neat. 

Tomorrow I'll open an issue to track fixing that. Thanks, @iquiw -- would *not* have guessed it worked that way :man_facepalming: